### ISSUE TITLE
AdaGC: Enhancing LLM Pretraining Stability via Adaptive Gradient Clipping

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#538]
## Links
https://openreview.net/forum?id=uBYlCu8b8Z

## Summary

損失スパイクは、大規模言語モデルの事前学習において依然として大きな障害となっています。これまでの研究では、個々の要因を調査することで損失スパイクの根本原因を特定しようと試みてきましたが、実際には、このようなスパイクは異質な要因が複合的に作用することで発生することが多いことがわかっています。経験的に、損失スパイクは、データ外れ値、ハードウェアまたは一時的な計算障害、数値精度の問題、ハイパーパラメータ設定などの組み合わせによって発生する可能性があります。根本的な原因に関わらず、これらのスパイクは不安定なオプティマイザ更新として現れ、異常な勾配が一次および二次モーメント状態の両方を汚染します。本論文では、勾配中心の原理に基づいた解決策として、AdaGCを提案します。AdaGCは、テンソルごとの勾配ノルムを、過去のクリップ値の指数移動平均に対して制限することで、このような汚染を軽減する適応型テンソルごとの勾配クリッピングスキームです。AdaGCはオプティマイザに依存せず、メモリオーバーヘッドはごくわずかで、特にハイブリッド並列分散学習において、GlobalGCと比較して通信コストを削減します。 Llama-2 7B、Mixtral 8×1B、およびERNIE 10B-A1.4Bを用いた実験では、AdaGCがトレーニングの不安定性を強力に排除し、すべてのモデルでスパイクスコアを一貫してゼロに低減し、GlobalGCと比較して下流の精度をそれぞれ1.32%、1.27%、2.48%向上させることが実証されました。さらに、AdaGCはMuonやLionなどのオプティマイザとシームレスに統合され、一貫して高い平均精度とゼロのスパイクスコアを実現します。

<img width="697" height="184" alt="Image" src="https://github.com/user-attachments/assets/ccb26c7d-34b0-41f7-9390-80f0189a0611" />

<img width="685" height="309" alt="Image" src="https://github.com/user-attachments/assets/f2026785-0a52-4233-9d9b-9e12f43d9af4" />

**Comment by Hiroki11x** [comment]
アルゴリズムの核は「grad norm の EMA を基準に clipping」**です。新しい最適化原理というより、単純なヒューリスティックを LLM pretraining 向けにうまく設計・検証した工学的研究
